Source/monster.cpp

```diff
@@ -844,22 +845,22 @@ void StartWalk2(int monsterId, int xvel, int yvel, int xoff, int yoff, int xadd,
 	dMonster[fx][fy] = monsterId + 1;
 	if (monster.mlid != NO_LIGHT)
 		ChangeLightXY(monster.mlid, monster.position.tile);
-	monster.position.offset = { xoff, yoff };
+	monster.position.offset = DisplacementOf<int16_t> { static_cast<int16_t>(xoff), static_cast<int16_t>(yoff) };
 	monster._mmode = MonsterMode::MoveSouthwards;
-	monster.position.velocity = { xvel, yvel };
+	monster.position.velocity = DisplacementOf<int16_t> { static_cast<int16_t>(xvel), static_cast<int16_t>(yvel) };
 	monster._mVar3 = static_cast<int>(endDir);
 	NewMonsterAnim(monster, MonsterGraphic::Walk, endDir, AnimationDistributionFlags::ProcessAnimationPending, -1);
-	monster.position.offset2 = { 16 * xoff, 16 * yoff };
+	monster.position.offset2 = DisplacementOf<int16_t> { static_cast<int16_t>(16 * xoff), static_cast<int16_t>(16 * yoff) };
 }
 
 void StartWalk3(int monsterId, int xvel, int yvel, int xoff, int yoff, int xadd, int yadd, int mapx, int mapy, Direction endDir)
 {
 	auto &monster = Monsters[monsterId];
 
-	int fx = xadd + monster.position.tile.x;
-	int fy = yadd + monster.position.tile.y;
-	int x = mapx + monster.position.tile.x;
-	int y = mapy + monster.position.tile.y;
+	const auto fx = static_cast<WorldTileCoord>(xadd + monster.position.tile.x);
+	const auto fy = static_cast<WorldTileCoord>(yadd + monster.position.tile.y);
+	const auto x = static_cast<WorldTileCoord>(mapx + monster.position.tile.x);
+	const auto y = static_cast<WorldTileCoord>(mapy + monster.position.tile.y);

```

Comment:
Would updating the attribute type not avoid the casting here?

Reply:
The xadd/mapx are `int` and can be negative, so it wouldn't by itself. Not sure if changing them to `int8_t` would but that's probably for another refactoring anyway.